### ISSUE TITLE
Bricked X2100

### ISSUE THREAD

**Issue by tiagoapmartins** [jwise/x2100-ec#12]
Hi everyone, it seems that my X2100 is bricked after a Fedora kernel update, I've attempted everything to get it booting again but I fear that it is indeed bricked. I've sent a message to @xy-tech regarding this and he said he would take a look at it provided that I sent him motherboard to china, but I wanted to see if it was possible to reflash the BIOS as I'm not very comfortable sending it back to china.
I've flashed libreboot to a x200 once with a raspberry pi, would it be a similar process?

As this machines are somewhat rare and special, I believe it is important to have this type of process detailed and available for everyone to see in the event someone runs into the same unfortunate issue. I wouldn't mind writing/detailing the process but I am a quite lost and would really appreciate any assistance.

**Comment by xy-tech** [comment]
Hi! Yea you can reflash the BIOS super easily with a RPI or a CH341a. Follow the flashing guide here: https://www.chucknemeth.com/laptop/lenovo-x230/flash-lenovo-x230-coreboot

I'd probably take some time to write a small guide on hardware flashing. It's similar in most machines actually since it's literally just reprogramming an EEPROM chip.

**Comment by exander77** [comment]
I can't get how the Fedora kernel upgrade would brick X2100. Something is not right.

**Comment by tiagoapmartins** [comment]
Well I certainly wasn't playing around with anything else. I had patched it with a version of this EC patch from this repo but it was months before this happening.

**Comment by exander77** [comment]
@RIPASSAS Nothing weird? Btw, do LEDs do anything? Do they lid on when you connect the charger?

**Comment by tiagoapmartins** [comment]
@exander77 I don't remember, I will have to assemble the machine again.

**Comment by jwise** [comment]
I also do not understand how a kernel upgrade would brick the machine.  If there are LEDs then the EC is alive.  If the main power LED turns on when you hit the power button then the CPU is at least trying to wake up.  You might have run into the ME defect, and your machine could be rerunning RAM training.  You should wait at least 30 seconds after hitting power to see if the screen lights up.

**Comment by xy-tech** [comment]
Worse case scenario is to reflash the machine externally. There should be some form of life at least. Very unlikely it’ll die like that.
________________________________

**Comment by exander77** [comment]
> I also do not understand how a kernel upgrade would brick the machine. If there are LEDs then the EC is alive. If the main power LED turns on when you hit the power button then the CPU is at least trying to wake up. You might have run into the ME defect, and your machine could be rerunning RAM training. You should wait at least 30 seconds after hitting power to see if the screen lights up.

It may not even be bricked.
1. I have that ME issue, and it takes 70 seconds to boot and may be dependent on the size of RAM.
2. One time when I disconnected power without battery, it refused to boot until I removed all the power including the BIOS battery.

**Comment by exander77** [comment]
I made this infographic a while back.
![CH341a-RaspberryPi-flash](https://user-images.githubusercontent.com/2256039/148386279-718acd63-571c-4bc4-952d-8a0f18a8c1f2.png)

**Comment by tiagoapmartins** [comment]
Hi everyone, thank you for all of the replies, I really appreciate it.
I have some good news, I've reassembled the machine and it suddenly started working again, so I guess it wasn't really bricked.
I do find very strange that when I first encountered this problem, I've unplugged every power source (including the CMOS battery), disconnected and reconnected all storage devices, Wi-Fi card and RAM sticks and it simply wasn't posting. I guess the solution is to leave it all unplugged for a while.
So @exander77 and @jwise judgement were correct regarding fedora kernel update not being able to brick the machine.
